backend/client-pkg/services/ModelImportService.js

```diff
@@ -37,18 +26,23 @@ const sendRequest = (requestPath, responseCallback) => {
     },
     withCredentials: false
   }, (res) => {
-    let responseObject = null;
+    var response = null;
 
     res.on('data', (bodyChunk) => {
-    if(!responseObject)
-      responseObject = bodyChunk;
-    else
-      responseObject += bodyChunk;
+      if(!response)
+        response = bodyChunk;
+      else
+        response += bodyChunk;
     });
 
     res.on('end', () => {
       try {
-        responseCallback(JSON.parse(responseObject));
+        if (response.includes('error')) {
```

Comment:
Validation should be on the response status code. It's possible that an error might not actually contain the word error or that a valid response could contain the word error (this would be really unlikely but still)